app/models.py

```diff
@@ -77,21 +77,20 @@ def key(self):
         return self.url
 
     def __eq__(self, other):
-        if isinstance(other, Resource):
-            if self.name != other.name:
-                return False
-            if self.url != other.url:
-                return False
-            if self.paid != other.paid:
-                return False
-            if self.notes != other.notes:
-                return False
-            if self.category != other.category:
-                return False
-            if self.languages != other.languages:
-                return False
-            return True
-        return False
+        return isinstance(other, Resource) and all(
+            (
+                getattr(self, attribute)
+                == getattr(other, attribute)
+                for attribute in [
```

Comment:
This looks like it might be less efficient because of the nested loop situation (I'm not sure how python actually interprets this, but it looks like a nested loop to me). However, in the interest of fixing the code complexity issue and because this looks more elegant and pythonic, I'm going to try to avoid premature optimization and go with it. Thanks 😄

Reply:
This is a good question! The outer loop is a generator, so it should only pull out the attributes from the inner loop 1 at a time. This means that if the first attribute test fails, the other attributes don't get tested as well.

Reply:
I'm open to coming back and optimizing this if needed in the future :)

Reply:
I think we're a little early to explore performance testing. Once we do, this might be an area we look at, but it all depends. For now, we need to get the front end coded up and possibly other clients like slackbot hooked into it before we'll have enough users to worry about performance.

Thanks for the offer 😄

Reply:
Ah, got it, just saw https://github.com/OperationCode/front-end/issues/661. Is there a public API that can be pinged to test out this resource without using the Slackbot?

Reply:
I'm not sure if you mean to say "Where is the resources API deployed?" But that's the question I'm going to answer. https://resources.operationcode.org has the docs, follow them to use the API

Reply:
Ah, despite my awkward phrasing, you figured out the question that I had meant to ask. Thanks!